db/migrate/20240208174825_add_indexes_to_companies_columns.rb

```diff
@@ -0,0 +1,7 @@
+class AddIndexesToCompaniesColumns < ActiveRecord::Migration[7.0]
+  def change
+    add_index :companies, :name
```

Comment:
Of course we can add composite indexes, I thought that for the basic use case, single indexes would do the work, and we don't have data about which fields will be used together to add index on.

Another thing that I thought about is adding a composite index for the 4 fields just for a proof of concept but I thought that this note will be a better idea to explain how I'm thinking about this problem.